pression/data/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+
+# Copyright (c) 2016 Stefan.Eilemann@epfl.ch
+
+set(PRESSIONDATA_PUBLIC_HEADERS
+  Compressor.h
+  CompressorInfo.h
+  Registry.h
+  types.h
+)
+
+set(SNAPPY_SOURCES
+  snappy/snappy.h
+  snappy/snappy.cc
+  snappy/snappy-sinksource.cc
+)
+
+set(ac_cv_have_stdint_h 1)
+set(ac_cv_have_stddef_h 1)
+set(ac_cv_have_sys_uio_h 0)
+set(SNAPPY_MAJOR 1)
+set(SNAPPY_MINOR 1)
+set(SNAPPY_PATCHLEVEL 3)
+configure_file(snappy/snappy-stubs-public.h.in
+  ${PROJECT_BINARY_DIR}/include/snappy-stubs-public.h @ONLY)
+
+if(CMAKE_COMPILER_IS_GCC OR CMAKE_COMPILER_IS_CLANG)
+  set_source_files_properties(${SNAPPY_SOURCES} PROPERTIES COMPILE_FLAGS
+    "-Wno-unused-parameter -Wno-header-hygiene -Wno-sign-compare")
+endif()
+
+set(PRESSIONDATA_COMPRESSORS
+  CompressorFastLZ.cpp
+  CompressorFastLZ.h
+  CompressorLZF.cpp
+  CompressorLZF.h
+  CompressorRLE.cpp
+  CompressorRLE.h
+  CompressorSnappy.cpp
+  CompressorSnappy.h
+  CompressorZSTD.cpp
+  CompressorZSTD.h
+  fastlz/fastlz.c
+  fastlz/fastlz.h
+  liblzf/lzf.h
+  liblzf/lzf_c.c
+  liblzf/lzf_d.c
```

Comment:
Can you fix the warnings compiling these files by either renaming the extension to .cpp or changing the compilation flags?